### PR TITLE
Add defensive stats to NCAAF teams

### LINKED ISSUE #193: Include defensive stats for NCAAF teams
**Is your feature request related to a problem? Please describe.**
I am writing an app that currently leverages the offensive team stats pulled from sportsreference - I would like to pull the defensive stats as well.

**Describe the solution you'd like**
As many defensive team stats (for a given year) as makes sense/are available should be returned alongside the offensive team stats.

**Additional context**
https://www.sports-reference.com/cfb/years/2019-team-defense.html

### PR DESCRIPTION
**Pull request by ajhodges**
- Added the following properties to a Team:
        `opponents_pass_completions`
        `opponents_pass_attempts`
        `opponents_pass_completion_percentage`
        `opponents_pass_yards`
        `opponents_interceptions`
        `opponents_pass_touchdowns`
        `opponents_rush_attempts`
        `opponents_rush_yards`
        `opponents_rush_yards_per_attempt`
        `opponents_rush_touchdowns`
        `opponents_plays`
        `opponents_yards`
        `opponents_turnovers`
        `opponents_fumbles_lost`
        `opponents_yards_per_play`
        `opponents_pass_first_downs`
        `opponents_rush_first_downs`
        `opponents_first_downs_from_penalties`
        `opponents_first_downs`
        `opponents_penalties`
        `opponents_yards_from_penalties`
- Added parsing logic for aforementioned fields
- Modified `_retrieve_all_teams` to merge defensive stats with conference/offensive stats.
- Added unit test assertions for Perdue's 2017 defensive stats

Closes #193

### CODE REVIEW

File: sportsreference/ncaaf/teams.py [hunk 1]
```diff
@@ -89,7 +111,7 @@ def _parse_team_data(self, team_data):
         """
         for field in self.__dict__:
             if field == '_year' or \
-               field == '_team_conference':
+                    field == '_team_conference':
```
Comment (roclark) [inline sportsreference/ncaaf/teams.py]: Just curious - was this padding intentional? If so, and you prefer this way, works for me! If not, would you be able to revert back to the original spacing?

File: sportsreference/ncaaf/teams.py [hunk 2]
```diff
@@ -419,13 +610,30 @@ def first_downs(self):
...
+        achieved per game.
+        """
+        return self._opponents_first_downs
+
     @float_property_decorator
     def penalties(self):
         """
-        Returns a ``float`` the average number of penalties conceded per game.
+        Returns a ``float`` of the average number of penalties conceded per
+        game.
```
Comment (roclark) [inline sportsreference/ncaaf/teams.py]: Similar to above - was this meant to be wrapped on a new line?

Comment (ajhodges) [reply sportsreference/ncaaf/teams.py]: The main change here was just the small grammar correction of adding `of` - I think I had to wrap the line so as not to exceed your line width of 79 chars

File: sportsreference/ncaaf/teams.py (hunk 1 above)
Comment (ajhodges) [reply sportsreference/ncaaf/teams.py]: Sorry about this one - I think I had PyCharm autoreformat the file or something. I'll revert this change.

File: sportsreference/ncaaf/teams.py (hunk 2 above)
Comment (roclark) [reply sportsreference/ncaaf/teams.py]: Ah yes, I completely missed that! Thanks for the change!

**Comment by roclark** [comment]
I noticed that your recent commit is now failing the Travis checks... Not really sure what happened since your changes are completely unrelated to the failing test, but I will dive into this and see why this is happening.

**Comment by roclark** [comment]
On further investigation, master is also failing, so the issue with the tests is definitely not related to your update. I will try and get a fix out to make sure this is passing, but otherwise I have no issue here and am happy to merge it! Thanks again for pushing this out!

**Comment by roclark** [comment]
Applied a fix and now everything is looking good here, so I will go ahead and merge! Thanks again for the great work! 🥇

**Pull request merged**